genie-agent/src/main/java/com/netflix/genie/agent/cli/JobRequestConverter.java

```diff
@@ -33,24 +38,30 @@
  */
 @Component
 @Lazy
-@NoArgsConstructor
 public class JobRequestConverter {
 
+    private final Validator validator;
+
+    JobRequestConverter(final Validator validator) {
```

Comment:
So I was curious and looked up where you can use `@Valid`. It looks like you can use it on return values from methods. So I'm wondering if you make the `agentJobRequestArgsToDTO` method have `@Valid` on the return if it will work as you desire here. It would save you from injecting the validator and invoking it directly. Also you'd probably have to mark the class `@Validated`. Cleaner code but downside is that validation would only trigger when this class was used in a spring context